evm/src/arithmetic/operations.rs

```diff
@@ -0,0 +1,156 @@
+use ethereum_types::U256;
+use plonky2::hash::hash_types::RichField;
+
+use crate::arithmetic::columns::*;
+use crate::arithmetic::{add, compare, modular, mul, sub};
+
+#[inline]
+fn u64_to_array<F: RichField>(out: &mut [F], x: u64) {
+    debug_assert!(out.len() == 4);
+
+    const MASK: u64 = (1 << 16) - 1;
+    out[0] = F::from_canonical_u64(x & MASK);
+    out[1] = F::from_canonical_u64((x >> 16) % MASK);
+    out[2] = F::from_canonical_u64((x >> 32) % MASK);
+    out[3] = F::from_canonical_u64((x >> 48) % MASK);
```

Comment:
Should these `%` be `&`? You can also get rid of `MASK` and use `F::from_canonical_u16(x as u16)` instead.